core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -91,7 +90,12 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
   /** List all the files currently stored on disk by the disk manager. */
   def getAllFiles(): Seq[File] = {
     // Get all the files inside the array of array of directories
-    subDirs.flatten.filter(_ != null).flatMap { dir =>
+    subDirs.flatMap { dir =>
```

Comment:
> This is simply a matter of program order, not multi-threading.

Could you clarify it? In multi-threading case, I think we should consider happens-before order rather than program order.